packages/rum-core/src/browser/performanceCollection.ts

```diff
@@ -109,8 +109,12 @@ export function startPerformanceCollection(lifeCycle: LifeCycle, configuration:
   })
 
   if (supportPerformanceObject()) {
-    handleRumPerformanceEntries(lifeCycle, configuration, performance.getEntries())
+    const performanceEntries = performance.getEntries()
+    // Because the performance entry list can be quite large
+    // delay the computation to prevent the SDK from blocking the main thread on init
+    setTimeout(() => handleRumPerformanceEntries(lifeCycle, configuration, performanceEntries))
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**❗ warning:** ‏we should monitor this function

Reply:
erf, of course. Done